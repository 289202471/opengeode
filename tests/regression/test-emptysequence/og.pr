SYSTEM og;
    /* CIF TEXT (159, 221), (289, 188) */
    -- Text area for declarations and comments

use dv comment 'dataview.asn';

signal dd;
signal we;
    /* CIF ENDTEXT */
    CHANNEL c
        FROM ENV TO og WITH dd;
        FROM og TO ENV WITH we;
    ENDCHANNEL;
    BLOCK og;
        SIGNALROUTE r
            FROM ENV TO og WITH dd;
            FROM og TO ENV WITH we;
        CONNECT c AND r;
        /* CIF PROCESS (225, 49), (150, 75) */
        PROCESS og;
            /* CIF TEXT (57, 58), (337, 130) */
            --  Test support for empty sequences and NULL type

dcl empty1 EmptySeq := c1;

dcl null1 NullType := NULL;

dcl choice1 ChoiceWithEmptySeq := a: n1;
            /* CIF ENDTEXT */
            /* CIF START (160, 226), (70, 35) */
            START;
                /* CIF TASK (137, 276), (115, 35) */
                TASK empty1 := c1
                /* CIF COMMENT (272, 276), (139, 35) */
                COMMENT 'Using a constant';
                /* CIF TASK (138, 326), (112, 35) */
                TASK empty1 := {}
                /* CIF COMMENT (270, 326), (135, 35) */
                COMMENT 'Classic notation';
                /* CIF TASK (130, 376), (130, 35) */
                TASK choice1 := b: c1;
                /* CIF TASK (131, 426), (126, 35) */
                TASK choice1 := b: {};
                /* CIF TASK (144, 476), (100, 35) */
                TASK null1 := n1;
                /* CIF TASK (134, 526), (121, 35) */
                TASK null1 :=  null;
                /* CIF NEXTSTATE (160, 576), (70, 35) */
                NEXTSTATE wait;
            /* CIF STATE (424, 231), (70, 35) */
            STATE wait;
                /* CIF INPUT (417, 286), (84, 35) */
                INPUT dd;
                    /* CIF OUTPUT (414, 336), (89, 35) */
                    OUTPUT we;
                    /* CIF NEXTSTATE (424, 386), (70, 35) */
                    NEXTSTATE wait;
            ENDSTATE;
        ENDPROCESS og;
    ENDBLOCK;
ENDSYSTEM;