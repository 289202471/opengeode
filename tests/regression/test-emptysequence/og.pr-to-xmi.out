[INFO] Starting to parse model ./test-emptysequence/og.pr
[ERROR] (visitSystem_definition) Unsupported child node:  AggregateNode (2 nodes) [nodes=[SignalNode [name=dd, param=null], SignalNode [name=we, param=null]]]
  	 Input: 'signal dd;' at pos 7:0
  	 File: ./test-emptysequence/og.pr
[ERROR] (visitPrim_expression_with_ID) Unsupported construct:  Input: 'a: n' at pos 28:34
  	 File: ./test-emptysequence/og.pr
[ERROR] Unsupported expression node: 
  	 NULL node
  	 Input: 'a: n' at pos 28:34
  	 File: ./test-emptysequence/og.pr
[ERROR] (visitPrim_empty_sequence) Unsupported construct:  Input: '{}' at pos 37:31
  	 File: ./test-emptysequence/og.pr
[ERROR] Unsupported expression node: 
  	 NULL node
  	 Input: '{}' at pos 37:31
  	 File: ./test-emptysequence/og.pr
[ERROR] (visitPrim_expression_with_ID) Unsupported construct:  Input: 'b: c' at pos 41:32
  	 File: ./test-emptysequence/og.pr
[ERROR] Unsupported expression node: 
  	 NULL node
  	 Input: 'b: c' at pos 41:32
  	 File: ./test-emptysequence/og.pr
[ERROR] (visitPrim_expression_with_ID) Unsupported construct:  Input: 'b: {}' at pos 43:32
  	 File: ./test-emptysequence/og.pr
[ERROR] Unsupported expression node: 
  	 NULL node
  	 Input: 'b: {}' at pos 43:32
  	 File: ./test-emptysequence/og.pr
[ERROR] (visitPrim_empty_sequence) Unsupported construct:  Input: '{}' at pos 37:31
  	 File: ./test-emptysequence/og.pr
[ERROR] Unsupported expression node: 
  	 NULL node
  	 Input: '{}' at pos 37:31
  	 File: ./test-emptysequence/og.pr
[ERROR] (visitPrim_expression_with_ID) Unsupported construct:  Input: 'b: c' at pos 41:32
  	 File: ./test-emptysequence/og.pr
[ERROR] Unsupported expression node: 
  	 NULL node
  	 Input: 'b: c' at pos 41:32
  	 File: ./test-emptysequence/og.pr
[ERROR] (visitPrim_expression_with_ID) Unsupported construct:  Input: 'b: {}' at pos 43:32
  	 File: ./test-emptysequence/og.pr
[ERROR] Unsupported expression node: 
  	 NULL node
  	 Input: 'b: {}' at pos 43:32
  	 File: ./test-emptysequence/og.pr
[FATAL] Failed to parse model
      java.lang.ClassCastException: ee.ibk.sdl.parser.sdl.nodes.StringNode cannot be cast to ee.ibk.sdl.parser.sdl.nodes.ActionNode

java.lang.ClassCastException: ee.ibk.sdl.parser.sdl.nodes.StringNode cannot be cast to ee.ibk.sdl.parser.sdl.nodes.ActionNode
	at ee.ibk.sdl.parser.sdl.SDLToGAVisitor.visitTransition(SDLToGAVisitor.java:1166)
	at ee.ibk.sdl.parser.sdl.SDLToGAVisitor.visitTransition(SDLToGAVisitor.java:1)
	at ee.ibk.sdl.parser.sdl.SDLParser$TransitionContext.accept(SDLParser.java:5533)
	at org.antlr.v4.runtime.tree.AbstractParseTreeVisitor.visit(AbstractParseTreeVisitor.java:18)
	at ee.ibk.sdl.parser.sdl.SDLToGAVisitor.visitInput_part(SDLToGAVisitor.java:1116)
	at ee.ibk.sdl.parser.sdl.SDLToGAVisitor.visitInput_part(SDLToGAVisitor.java:1)
	at ee.ibk.sdl.parser.sdl.SDLParser$Input_partContext.accept(SDLParser.java:5257)
	at org.antlr.v4.runtime.tree.AbstractParseTreeVisitor.visitChildren(AbstractParseTreeVisitor.java:46)
	at ee.ibk.sdl.parser.sdl.SDLToGAVisitor.visitState_part(SDLToGAVisitor.java:1026)
	at ee.ibk.sdl.parser.sdl.SDLToGAVisitor.visitState_part(SDLToGAVisitor.java:1)
	at ee.ibk.sdl.parser.sdl.SDLParser$State_partContext.accept(SDLParser.java:4547)
	at org.antlr.v4.runtime.tree.AbstractParseTreeVisitor.visit(AbstractParseTreeVisitor.java:18)
	at ee.ibk.sdl.parser.sdl.SDLToGAVisitor.visitState(SDLToGAVisitor.java:916)
	at ee.ibk.sdl.parser.sdl.SDLToGAVisitor.visitState(SDLToGAVisitor.java:1)
	at ee.ibk.sdl.parser.sdl.SDLParser$StateContext.accept(SDLParser.java:3397)
	at org.antlr.v4.runtime.tree.AbstractParseTreeVisitor.visit(AbstractParseTreeVisitor.java:18)
	at ee.ibk.sdl.parser.sdl.SDLToGAVisitor.visitProcessBody(SDLToGAVisitor.java:849)
	at ee.ibk.sdl.parser.sdl.SDLToGAVisitor.visitProcessBody(SDLToGAVisitor.java:1)
	at ee.ibk.sdl.parser.sdl.SDLParser$ProcessBodyContext.accept(SDLParser.java:3105)
	at org.antlr.v4.runtime.tree.AbstractParseTreeVisitor.visit(AbstractParseTreeVisitor.java:18)
	at ee.ibk.sdl.parser.sdl.SDLToGAVisitor.visitProcess_definition(SDLToGAVisitor.java:530)
	at ee.ibk.sdl.parser.sdl.SDLToGAVisitor.visitProcess_definition(SDLToGAVisitor.java:1)
	at ee.ibk.sdl.parser.sdl.SDLParser$Process_definitionContext.accept(SDLParser.java:1175)
	at org.antlr.v4.runtime.tree.AbstractParseTreeVisitor.visit(AbstractParseTreeVisitor.java:18)
	at ee.ibk.sdl.parser.sdl.SDLToGAVisitor.visitBlock_definition(SDLToGAVisitor.java:456)
	at ee.ibk.sdl.parser.sdl.SDLToGAVisitor.visitBlock_definition(SDLToGAVisitor.java:1)
	at ee.ibk.sdl.parser.sdl.SDLParser$Block_definitionContext.accept(SDLParser.java:871)
	at org.antlr.v4.runtime.tree.AbstractParseTreeVisitor.visitChildren(AbstractParseTreeVisitor.java:46)
	at ee.ibk.sdl.parser.sdl.SDLToGAVisitor.visitEntity_in_system(SDLToGAVisitor.java:344)
	at ee.ibk.sdl.parser.sdl.SDLToGAVisitor.visitEntity_in_system(SDLToGAVisitor.java:1)
	at ee.ibk.sdl.parser.sdl.SDLParser$Entity_in_systemContext.accept(SDLParser.java:568)
	at org.antlr.v4.runtime.tree.AbstractParseTreeVisitor.visit(AbstractParseTreeVisitor.java:18)
	at ee.ibk.sdl.parser.sdl.SDLToGAVisitor.visitSystem_definition(SDLToGAVisitor.java:275)
	at ee.ibk.sdl.parser.sdl.SDLToGAVisitor.visitSystem_definition(SDLToGAVisitor.java:1)
	at ee.ibk.sdl.parser.sdl.SDLParser$System_definitionContext.accept(SDLParser.java:360)
	at org.antlr.v4.runtime.tree.AbstractParseTreeVisitor.visit(AbstractParseTreeVisitor.java:18)
	at ee.ibk.sdl.parser.sdl.SDLToGAVisitor.visitPr_file(SDLToGAVisitor.java:155)
	at ee.ibk.sdl.parser.sdl.SDLToGAVisitor.visitPr_file(SDLToGAVisitor.java:1)
	at ee.ibk.sdl.parser.sdl.SDLParser$Pr_fileContext.accept(SDLParser.java:276)
	at org.antlr.v4.runtime.tree.AbstractParseTreeVisitor.visit(AbstractParseTreeVisitor.java:18)
	at ee.ibk.sdl.parser.ParseHandler.doParseSDL(ParseHandler.java:64)
	at ee.ibk.sdl.parser.ParseHandler.parseFile(ParseHandler.java:139)
	at ee.ibk.sdl.parser.Main.processFile(Main.java:588)
	at ee.ibk.sdl.parser.Main.processInputFiles(Main.java:485)
	at ee.ibk.sdl.parser.Main.doMain(Main.java:190)
	at ee.ibk.sdl.parser.Main.main(Main.java:132)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.eclipse.jdt.internal.jarinjarloader.JarRsrcLoader.main(JarRsrcLoader.java:58)
java.lang.RuntimeException: Fatal error occurred. The tool will stop now.
	at ee.ibk.sdl.parser.utils.EventHandler.handle(EventHandler.java:141)
	at ee.ibk.sdl.parser.utils.EventHandler.fatal(EventHandler.java:171)
	at ee.ibk.sdl.parser.Main.processFile(Main.java:593)
	at ee.ibk.sdl.parser.Main.processInputFiles(Main.java:485)
	at ee.ibk.sdl.parser.Main.doMain(Main.java:190)
	at ee.ibk.sdl.parser.Main.main(Main.java:132)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.eclipse.jdt.internal.jarinjarloader.JarRsrcLoader.main(JarRsrcLoader.java:58)
